public/js/custom-analytics.js

```diff
@@ -0,0 +1,150 @@
+jQuery(function($) {
+
+  // Debug flag
+  var debugMode = false;
+
+  // Default time delay before checking location
+  var callBackTime = 100;
+
+  // # px before tracking a reader
+  var readerLocation = 150;
+
+  // Set some flags for tracking & execution
+  var timer = 0;
+  var scroller = false;
+  var endOpeningTimes = false;
+  var endPatientsSay = false;
+  var endSurgeryGPs = false;
+  var endServices = false;
+  var endParking = false;
+  var didComplete = false;
+
+  // Set some time variables to calculate reading time
+  var startTime = new Date();
+  var beginning = startTime.getTime();
+  var totalTime = 0;
+
+  // Get some information about the current page
+  var pageTitle = document.title;
+
+  // Track the aticle load
+  if (!debugMode) {
+      //ga('send', 'event', 'Reading', 'ArticleLoaded', '');
+
+  } else {
+      console.log('Start custom analytics');
+  }
+
+  // Check the location and track user
+  function trackLocation() {
+      bottom = $(window).height() + $(window).scrollTop();
+      height = $(document).height();
+      results = $('.reading-width .sr-only').attr('data-results');
+      results_14 = Math.round(results / 4);
+      results_12 = Math.round(results / 2);
+      results_34 = Math.round(results_14 * 3);
+
+
+      // If user starts to scroll send an event
+      if (bottom > readerLocation && !scroller) {
+          currentTime = new Date();
+          scrollStart = currentTime.getTime();
+          timeToScroll = Math.round((scrollStart - beginning) / 1000);
+          if (!debugMode) {
+              console.log('started reading ' + timeToScroll);
+              ga('send', 'event', 'User started scrolling', timeToScroll+'s');
+          } else {
+              alert('User started scrolling (Time: ' + timeToScroll + 's)');
+          }
+          scroller = true;
+      }
+
+      if (bottom > $('h2.opening-times:contains("GP")').offset().top + $('h2.opening-times:contains("GP")').outerHeight() && !endOpeningTimes) {
+          currentTime = new Date();
+          openingTimesScrollEnd = currentTime.getTime();
+          timeToOpeningTimesEnd = Math.round((openingTimesScrollEnd - scrollStart) / 1000);
+          if (!debugMode) {
+              ga('send', 'event', 'User reached the "GP Opening Times" section', timeToOpeningTimesEnd+'s');
+          } else {
+              alert('User reached the "GP Opening Times" section (Time: ' + timeToOpeningTimesEnd + 's)');
+          }
+          endOpeningTimes = true;
```

Comment:
There's a lot of repeated blocks of similar code, that could be extracted into a function and called with different parameters, something along the lines of
`recordScrolledPassed('h2:contains("Parking")', 'User reached the "Parking and accessibility" section', endParking)`